network/network.go

```diff
@@ -633,3 +634,18 @@ func FormatAsCIDR(addresses []string) ([]string, error) {
 	}
 	return result, nil
 }
+
+// macAddressTemplate is suitable for generating virtual MAC addresses,
+// particularly for use by container devices.
+// The last 3 segments are randomised.
+const macAddressTemplate = "00:16:3e:%02x:%02x:%02x"
+
+// GenerateVirtualMACAddress creates a random MAC address within the address
+// space implied by macAddressTemplate above.
+var GenerateVirtualMACAddress = func() string {
```

Comment:
Should we keep the TODO about uniqueness in the current space?

Reply:
I will restore it.

Reply:
Birthday paradox says that we have a 50% probability to hit a collision if we generate 65536^1/2 = 255 addresses.
https://en.wikipedia.org/wiki/Birthday_problem

That does start to sound like something that could be hit in large openstack deploys... :(
This should probably be a bug that we explicitly track.